### PR TITLE
fix the dimension checking logic

### PR DESCRIPTION
**Pull request by Cadair**
A very small bug fix I found

### CODE REVIEW

File: ndcube/ndcube.py [hunk 1]
```diff
@@ -405,7 +405,7 @@ def crop_by_coords(self, min_coord_values, interval_widths):
         # The docstring is defined in NDDataBase
 
         n_dim = len(self.dimensions)
-        if len(min_coord_values) != len(interval_widths) != n_dim:
+        if (len(min_coord_values) != len(interval_widths)) or len(min_coord_values) != n_dim:
```
Comment (DanRyanIrish) [inline ndcube/ndcube.py]: What's the difference in the logic of these two implementations?

Comment (Cadair) [reply ndcube/ndcube.py]: one works the other dosen't :laughing: 

I can't immediately work out why the first one dosen't work

Comment (DanRyanIrish) [reply ndcube/ndcube.py]: But ```NDCube.dimensions``` doesn't crash.  Do you mean that this if statement will never be entered or at least isn't entered when it should?

Comment (Cadair) [reply ndcube/ndcube.py]:
```python
>>> d1 = 3
>>> d2 = 3
>>> nd = 4
>>> d1 != d2 != nd
False
>>> nd = 3
>>> d1 != d2 != nd
```

Comment (DanRyanIrish) [reply ndcube/ndcube.py]: Bizarre!

**Comment by DanRyanIrish** [comment]
lgtm pending Travis.

**Pull request merged**